topy/topy.py

```diff
@@ -121,9 +121,22 @@ def sanitize_filename(filename):
 def print_diff(filename, old, new, stream=sys.stdout):
     """Diffs the `old` and `new` strings and prints as unified diff to file-like object `stream`."""
 
-    # TODO: color output for terminals
     lines = unified_diff(old.splitlines(True), new.splitlines(True), filename, filename)
-    stream.writelines(lines)
+    lines_color = add_output_color(lines)
```

Comment:
Note that a colored diff can no longer be used by `patch`, `git`, etc commands.

So I think there should be a command line option (`--color=`) to disable/enable colors. By default you should auto-detect, and enable colors only outputting directly to a terminal. This behavior almost universal in Unix tools:
```
diff --help
      --color[=WHEN]       colorize the output; WHEN can be 'never', 'always',
                             or 'auto' (the default)

ls --help
      --colo[u]r=WHEN    when to use terminal colours (always, auto, never)

grep --help
      --color[=WHEN],
      --colour[=WHEN]       use markers to highlight the matching strings;
                            WHEN is 'always', 'never', or 'auto'
```

A simple way to detect terminal output is: `terminal_output = hasattr(stream, 'isatty') and stream.isatty()`

Reply:
Thanks, that's great feedback - I'll look into this